version: "2.4"

services:
  web:
    build:
      context: .
      dockerfile: Dockerfile-python
    restart: on-failure
    depends_on:
      - postgres
    ports:
      - "1000:4000"
    volumes:
      - ./script:/app
    environment:
      - POSTGRES_HOST=postgres
      - POSTGRES_DB=geovisualizer
      - POSTGRES_PASSWORD=
      - POSTGRES_USER=postgres
    command:
      ["./app/wait-for-postgres.sh", "postgres", "python", "./app/api.py"]

  python:
    build:
      context: .
      dockerfile: Dockerfile-python
    restart: on-failure
    depends_on:
      - postgres
    volumes:
      - ./script:/app
    environment:
      - MAPBOX_ACCESS_TOKEN=${MAPBOX_ACCESS_TOKEN}
      - MAPBOX_HOST=api.mapbox.com
      - POSTGRES_HOST=postgres
      - POSTGRES_DB=geovisualizer
      - POSTGRES_PASSWORD=
      - POSTGRES_USER=postgres
    command:
      ["./app/wait-for-postgres.sh", "postgres", "python", "./app/poi.py"]

  postgres:
    build:
      context: .
      dockerfile: Dockerfile-postgres
    restart: on-failure
    ports:
      - "6432:5432"
    volumes:
      - ./sql:/opt/sql
      - ./docker/postgres:/opt/refresh
      - postgres-data:/var/lib/postgresql/data
volumes:
  postgres-data:
